src/groupeddataframe/groupeddataframe.jl

```diff
@@ -387,6 +391,29 @@ function Base.to_index(gd::GroupedDataFrame, key::NamedTuple{N}) where {N}
     return Base.to_index(gd, Tuple(key))
 end
 
+function _dict_to_tuple(key::AbstractDict{<:AbstractString} , gd)
+    T = keytype(key)
+    return ntuple(length(gd.cols)) do i 
+        # Most `<: AbstractString` types do not support constructors from symbols
+        key[T(string(gd.cols[i]))]
```

Comment:
I would assume this should be enough (If I am right then capturing `T` above is not needed)
```suggestion
        key[string(gd.cols[i])]
```
also why do you prefer `do` syntax here instead of just `i -> key[string(gd.cols[i])]`?

Reply:
It was longer before for some reason. Made in line now.